### PR TITLE
Various small fixes

### PR DESCRIPTION
**Pull request by dlyr**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Mostly fixes + precompiled headers

If needed I can split this PR in small parts (e.g. if some point need discussion, I'll split the ok part to a first PR, and we can discuss the rest).

* **Behavior changes**

Add getter for AttribArrayDisplayable's vao and vbo OpenGL's handle (e.g. for cudo interoperability).

Clean lambda capture (remove warning, also present in #950 ...) 

4b1de5209a [engine] fix forward decl RenderParameters
One miss forward declared class

8434cd6528 [engine] unregister component messaging on delete.
When removing a component, also unregister corresponding registered calls 

3d74db8a2b [engine] Add gl namespace to GL_ macros.
usefull to prevent name clash in some cases.

a3e3609b56 [gui] add missing pragma in RotateAroundCameraManip include
One header do not have it's pragma once

17642d88a6 [gui] Add default switch in TimelineScrollArea key event.
to suppress a warning

b21753c3a0 [gui] BaseApplication don't call radiumFrame when isAboutToQuit.
which crash on quit on windows 

0f36ca94b8 [gui] move Radium menu and about action to it's own function.
In case your application don't want to have this about radium menu

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
Need to update application code to call addRadiumMenu()

### CODE REVIEW

**Comment by MathiasPaulin** [comment]
I'm not sure that integrating precompiled_headers in this PR is a good choice. I agree with the others but I'm dubious about this.

**Comment by dlyr** [comment]
Fine for me, I have splitted precompiled header to #961

File: src/Engine/Data/BlinnPhongMaterial.inl [hunk 1]
```diff
@@ -1,9 +1,12 @@
 #pragma once
 
+#include "BlinnPhongMaterial.hpp"
```
Comment (MathiasPaulin) [inline src/Engine/Data/BlinnPhongMaterial.inl:3]:
```suggestion
#include <Engine/Data/BlinnPhongMaterial.hpp>
```

File: src/Engine/Scene/ComponentMessenger.cpp [hunk 2]
```diff
@@ -1,11 +1,30 @@
...
+        if ( listItr != entityList.get().end() ) {
+            CallbackMap& list = listItr->second;
+            auto pred = [comp]( const Key& k ) -> bool { return k.first == comp->getName(); };
+            for ( auto it = list.begin(); it != list.end(); ) {
+                if ( pred( it->first ) )
+                    it = list.erase( it );
+                else
+                    ++it;
+            }
+        }
```
[the remark this answers is not in the document]
Comment (dlyr) [reply src/Engine/Scene/ComponentMessenger.cpp:24]: Damn git that do not comment when I click resolve.
My first tests shows that remove_if is not compatible with std::map, we need erase_if from c++20.
If you find a cleaner solution, update welcome.
(If not, I can add a comment to say it's to reproduce remove_if behavior).

[the remark this answers is not in the document]
Comment (MathiasPaulin) [reply src/Engine/Scene/ComponentMessenger.cpp:24]: I think such a comment about replacement of this code by C++20 std::erase_if (unsupported by radium right now) and referring to https://en.cppreference.com/w/cpp/container/unordered_map/erase_if as sample implementation, could be helpful.

File: src/Engine/Data/Mesh.hpp [hunk 3]
```diff
@@ -113,6 +113,14 @@ class RA_ENGINE_API AttribArrayDisplayable : public Displayable
     virtual Core::Geometry::AttribArrayGeometry& getAttribArrayGeometry()             = 0;
     ///@}
 
+    /// get opengl's vbo handle (uint) corresponding to attrib \b name.
```
Comment (MathiasPaulin) [inline src/Engine/Data/Mesh.hpp:116]:
```suggestion
    /// \brief Get opengl's vbo handle (uint) corresponding to attrib \b name.
```

File: src/Engine/Data/Mesh.hpp [hunk 4]
```diff
@@ -113,6 +113,14 @@ class RA_ENGINE_API AttribArrayDisplayable : public Displayable
...
+    /// if vbo is not initialized or name do not correponds to an actual attrib name, the returned
```
Comment (MathiasPaulin) [inline src/Engine/Data/Mesh.hpp:117]:
```suggestion
    /// If vbo is not initialized or name do not correponds to an actual attrib name, the returned
```

File: src/Engine/Data/Mesh.hpp [hunk 5]
```diff
@@ -113,6 +113,14 @@ class RA_ENGINE_API AttribArrayDisplayable : public Displayable
...
+    /// optional is empty
+    Ra::Core::Utils::optional<gl::GLuint> getVboHandle( const std::string& name );
+    /// get opengl's vao handle (uint)
```
Comment (MathiasPaulin) [inline src/Engine/Data/Mesh.hpp:120]:
```suggestion
    /// \brief Get opengl's vao handle (uint)
```

**Review by MathiasPaulin — APPROVED** [review]

**Pull request merged**